tests/xharness/Execution/Mlaunch/Arguments.cs

```diff
@@ -90,4 +126,165 @@ public ListExtraDataArgument () : base ("list-extra-data")
 		{
 		}
 	}
+
+	/// <summary>
+	/// Attach native debugger.
+	/// </summary>
+	public sealed class AttachNativeDebuggerArgument : OptionArgument {
+		public AttachNativeDebuggerArgument () : base ("attach-native-debugger")
+		{
+		}
+	}
+	
+	/// <summary>
+	/// Attempt to disable memory limits for launched apps.
+	/// This is just an attempt, some or all usual limits may still be enforced.
+	/// </summary>
+	public sealed class DisableMemoryLimitsArgument : OptionArgument {
+		public DisableMemoryLimitsArgument () : base ("disable-memory-limits")
+		{
+		}
+	}
+
+	/// <summary>
+	/// 
+	/// </summary>
+	public sealed class WaitForExitArgument : OptionArgument {
+		public WaitForExitArgument () : base ("wait-for-exit")
+		{
+		}
+	}
+
+	/// <summary>
+	/// Launch the app with this command line argument. This must be specified multiple times for multiple arguments.
+	/// </summary>
+	public sealed class SetAppArgumentArgument : MlaunchArgument {
+		readonly string value;
+
+		public SetAppArgumentArgument (string value, bool isAppArg = false)
+		{
+			this.value = value ?? throw new System.ArgumentNullException (nameof (value));
+
+			if (isAppArg)
+				this.value = "-app-arg:" + this.value;
+		}
+
+		public override string AsCommandLineArgument () => "-argument=" + value;
+	}
+
+	/// <summary>
+	/// Set the environment variable in the application on startup.
+	/// </summary>
+	public sealed class SetEnvVariableArgument : MlaunchArgument {
+		readonly string variableName;
+		readonly string variableValue;
+
+		public SetEnvVariableArgument (string variableName, object variableValue)
+		{
+			this.variableName = variableName ?? throw new System.ArgumentNullException (nameof (variableName));
+			this.variableValue = variableValue?.ToString() ?? throw new System.ArgumentNullException (nameof (variableValue));
+
+			if (variableValue is bool)
+				this.variableValue = this.variableValue.ToLower ();
```

Comment:
`ToLowerInvariant`
otherwise we depend on bot config for the result

Reply:
Will handle in a follow-up PR and merge this to unblock the pipeline

Reply:
Fixed here: https://github.com/xamarin/xamarin-macios/pull/8197